include/ipr/impl

```diff
@@ -1258,10 +1258,8 @@ namespace ipr {
                                        const ipr::Expr_list&);
          Static_cast* make_static_cast(const ipr::Type&, const ipr::Expr&);
          Binary_fold* make_binary_fold(Category_code, const ipr::Expr&, const ipr::Expr&, Optional<ipr::Type> = {});
-         New* make_new(const ipr::Type&, Optional<ipr::Type> = {});
-         New* make_new(const ipr::Type&, const ipr::Expr_list&, Optional<ipr::Type> = {});
-         New* make_new(const ipr::Expr_list&, const ipr::Type&,
-                       const ipr::Expr_list&, Optional<ipr::Type> = {});
+         New* make_new(const ipr::Type& allocated, Optional<ipr::Expr_list> init = {},
+                       Optional<Expr_list> placement = {}, Optional<ipr::Type> result = {});
          Conditional* make_conditional(const ipr::Expr&, const ipr::Expr&,
```

Comment:
This new function now has two sam parameter types next to each other.

Reply:
What parameter order do you suggest?

I cannot move allocated type back as it is non-optional. If I move the result type forward, it will be the only function where result type is not the last parameter.

Reply:
The new parameter types order is good.